main.ts

```diff
@@ -1,234 +1,334 @@
-
-	}
-
-	onunload() {
-
-		console.debug('unloading plugin');
-	}
-
-	async loadSettings() {
-		this.settings = Object.assign({}, DEFAULT_SETTINGS, await this.loadData());
-	}
-
-	async saveSettings() {
-		await this.saveData(this.settings);
-	}
-
-	updateIndexContent = async (changedFile: string): Promise<void> => {
-		await this.generateIndexContents(changedFile);
-	}
-
-
-	generateIndexContents = async (indexFile: string): Promise<void> => {
-
-		let indexTFile = this.app.vault.getAbstractFileByPath(indexFile) || await this.app.vault.create(indexFile, '');
-
-		if (indexTFile && indexTFile instanceof TFile)
-			return this.generateIndexContent(indexTFile);
-	}
-
-	generateIndexContent = async (indexTFile: TFile): Promise<void> => {
-
-		const indexContent = indexTFile.parent.children
-			.reduce(
-				(acc, curr) => {
-					acc.push(this.generateIndexItem(curr.path))
-					return acc;
-				}, []);
-		const parentLink = this.getParentFolder(indexTFile.path)
-		if (parentLink && parentLink !== '') {
-			indexContent.push(this.generateIndexItem(parentLink));
-		}
-		try {
-			if (indexTFile instanceof TFile)
-				await this.app.vault.modify(indexTFile, indexContent.join('\n'));
-			else {
-				throw new Error('Creation index as folder is not supported');
-			}
-		} catch (e) {
-			console.warn('Error during deletion/creation of index files');
-		}
-	}
-
-	generateIndexItem = (path: string): string => {
-
-		switch(this.settings.intexItemStyle){
-			case IndexItemStyle.PureLink:
-				return `[[${path}]]`
-			case IndexItemStyle.List:
-				return `- [[${path}]]`
-			case IndexItemStyle.Checkbox:
-				return `- [ ] [[${path}]]`
-		};
-	}
-	getIndexFilePath = (filePath: string): string => {
-
-		const fileAbstrPath = this.app.vault.getAbstractFileByPath(filePath);
-
-
-		if (this.isIndexFile(fileAbstrPath))
-			return null;
-		let parentPath = this.getParentFolder(filePath);
-
-		// if its parent does not exits, then its a moved subfolder, so it should not be updated
-		const parentTFolder = this.app.vault.getAbstractFileByPath(parentPath);
-		if (parentPath && parentPath !== '') {
-			if (!parentTFolder)
-				return undefined;
-			parentPath = `${parentPath}/`;
-		}
-		const parentName = this.getParentFolderName(filePath);
-
-		return `${parentPath}${this.settings.indexPrefix}${parentName}.md`;
-
-	}
-
-	getParentFolder = (filePath: string): string => {
-		const fileFolderArray = filePath.split('/');
-		fileFolderArray.pop();
-
-		return fileFolderArray.join('/');
-	}
-
-	getParentFolderName = (filePath: string): string => {
-		const parentFolder = this.getParentFolder(filePath);
-		const fileFolderArray = parentFolder.split('/');
-		return (fileFolderArray[0] !== '') ? fileFolderArray[fileFolderArray.length - 1] : this.app.vault.getName();
-	}
-
-	isIndexFile = (file: TAbstractFile): boolean => {
-
-		return file instanceof TFile && file.name.startsWith(this.settings.indexPrefix);
-
-	}
-
+  settings: ZoottelkeeperPluginSettings;

```

Comment:
I think the lint config is different , because the tabs are replaced by double-spaces, and therefore it's really hard to see what are the real changes. Could you please change the double-spaces indentations to tabs? Thank you!

Reply:
Oh! Yes, I'll change that. Should I make a new PR afterwards? (Sorry if that's a stupid question, it's my first 😅)

Reply:
Nope, this PR should be updated automatically if you push your changes to the forked master. There are no stupid questions btw :)